schema-engine/connectors/schema-connector/src/warnings.rs

```diff
@@ -31,19 +31,18 @@ fn display_list<T: Ord>(
     project_field: fn(&T) -> &str,
     f: &mut fmt::Formatter<'_>,
 ) -> fmt::Result {
-    let sorted: BTreeSet<_> = items.iter().collect();
-    let mut sorted = sorted.into_iter().peekable();
+    let mut items = items.iter().peekable();
     let mut key = None;
     let close = |f: &mut fmt::Formatter<'_>| f.write_str("]\n");
 
-    while let Some(next) = sorted.next() {
+    while let Some(next) = items.next() {
```

Comment:
The reason we sort is because different warnings on the same model can come at different points during introspection, so they would be split into two groups in the rendered output. We can fix that by using a custom sort order, but we have to sort.

Reply:
If we stop using the BTreeSet here, I'd also remove the `Eq`, `PartialOrd` and `Ord` derives that were added in the last PR.

Reply:
> The reason we sort is because different warnings on the same model can come at different points during introspection, so they would be split into two groups in the rendered output. We can fix that by using a custom sort order, but we have to sort.

Would you have a test case that would add different warnings on the same model in "different points during introspection", in a way that would have to require sorting?

Reply:
I'll have to spend time reading the introspection implementation again to do that, but I'm very confident that this is broken.

Reply:
So I spent the time to check, and _at the moment_, introspection happens in an order where the warnings for the same model will always be contiguous. That can still break easily, and we should sort. Something like:

```rust
let mut items = items.to_owned();
items.sort_by_key(|item| project_key(item));
let items = items.iter().peekable();
// ...
```

Reply:
Done, tests still pass locally